java/src/org/openqa/selenium/devtools/CdpEndpointFinder.java

```diff
@@ -91,9 +91,11 @@ public static Optional<URI> getReportedUri(Capabilities caps) {
         break;
       case "firefox":
         key = "moz:debuggerAddress";
-        LOG.warning(
-            "CDP support for Firefox is deprecated and will be removed in future versions. "
-                + "Please switch to WebDriver BiDi.");
+        if (!(Boolean) caps.getCapability("webSocketUrl")) {
```

Comment:
Wouldn't this throw if the capability is not present? Maybe we check first if it is present and then we check the boolean value.

Reply:
You mean to do a `null` check first?

Reply:
You can do that. But I just saw that `caps.is("webSocketUrl");` should do the trick.

Reply:
Done